### PR TITLE
Rename at-scriptdir project argument to at-script and search upwards for Project.toml

### LINKED ISSUE #53352: Rename @scriptdir to @script, and look for environments in parent directories
In today's triage call, Stefan's comment about `@scriptdir` https://github.com/JuliaLang/julia/pull/51149#issuecomment-1938993314 (introduced in #50864) was discussed a bit further, and triage seemed broadly approving of the idea that:
- `@scriptdir` be renamed to `@script`
- `@script` behaves like `@.`, in that it automatically searches up parent directories for the first environment found

These changes (specifically the first one) will need to be made before 1.11 to be non-breaking.

### PR DESCRIPTION
**Pull request by oxinabox**
Closes https://github.com/JuliaLang/julia/issues/53352

I also noticed it wasn't mentioned in the NEWs.md
and so I added it .

Reconciling the relative path behavior, with the search upwads behavour requested in #53352 is a thing.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -68,6 +68,7 @@ difference between defining a `main` function and executing the code directly at
 * The `--compiled-modules` and `--pkgimages` flags can now be set to `existing`, which will
   cause Julia to consider loading existing cache files, but not to create new ones ([#50586]
   and [#52573]).
+* The `--project` argument now accepts `@script` to give a path to a directory with a Project.toml relative to the passed script file. `--project=@script/foo` for the `foo` subdirectory. If no path is given after (i.e. `--project=@script`) then (like `--project=@`) the directory and its parents are searched for a Project.toml ([#50864] and [#53352])
```
Comment (oxinabox) [inline NEWS.md:71]:
```suggestion
* The `--project` argument now accepts `@script` to give a path to a directory with a Project.toml relative to the passed script file. `--project=@script/foo` for the `foo` subdirectory. If no path is given after (i.e. `--project=@script`) then (like `--project=@.`) the directory and its parents are searched for a Project.toml ([#50864] and [#53352])
```

**Comment by KristofferC** [comment]
Would be good with a rebase + test. The NEWS entry should probably be moved to HISTORY as well.

File: base/initdefs.jl [hunk 2]
```diff
@@ -283,7 +283,12 @@ function load_path_expand(env::AbstractString)::Union{String, Nothing}
                 end
                 dir = dirname(ARGS[1])
             end
-            return abspath(replace(env, "@scriptdir" => dir))
+            if env == "@script"  # complete match, not startswith, so search upwards
+                return current_project(dir)
+            else
+                # starts with, so assume rleative path is after
```
Comment (KristofferC) [inline base/initdefs.jl:289]: "rleative"

**Comment by oxinabox** [comment]
This needs to be put into the release-1.11 branch.
Or it will be a breaking change.
(I made this after https://github.com/JuliaLang/julia/issues/53352 before the 1.11 branch was cut specifically to avoid that)

File: base/initdefs.jl (hunk 2 above)
Comment (oxinabox) [reply base/initdefs.jl:289]:
```suggestion
                # starts with, so assume relative path is after
```

**Comment by KristofferC** [comment]
> Since that branch has been cut i am not sure how to make this PR right for porting

Just move the NEWS entry to HISTORY and rebase. Should be good to be backported after that?

**Comment by KristofferC** [comment]
Rebased.

**Pull request merged**